packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -337,7 +350,27 @@ const Panel: React.FC<SelectPanelProps> = ({
 }
 
 const SelectPanelButton = React.forwardRef<HTMLButtonElement, ButtonProps>((props, anchorRef) => {
-  return <Button ref={anchorRef} {...props} />
+  const inputProps = useFormControlForwardedProps(props)
+  const [labelText, setLabelText] = useState('')
+  useEffect(() => {
+    const label = document.querySelector(`[for='${inputProps.id}']`)
+    if (label?.textContent) {
+      setLabelText(label.textContent)
+    }
+  }, [inputProps.id])
+
+  if (labelText && inputProps.children) {
+    return (
+      <Button
+        ref={anchorRef}
+        aria-label={`${labelText}, ${(anchorRef as MutableRefObject<HTMLButtonElement>).current.textContent}`}
```

Comment:
Could we flip this, so that the button text content is first?

```suggestion
        aria-label={`${(anchorRef as MutableRefObject<HTMLButtonElement>).current.textContent}, ${labelText}`}
```

This way, the contents of the button is announced first, and the label second. We'd want this mainly because the most important content is the button's text content, whereas the label itself would be secondary to that, after it has been read.

Reply:
Sure I can go ahead and flip that 👍

Reply:
Changes are pushed